### PR TITLE
[NES-263] [N3] [Medium] Wrong asset check when async minting

### PR DESCRIPTION
**Pull request by ungaro**
## What's new in this PR?

perform checks using `sharesDepositRequest` directly

### CODE REVIEW

File: nest/src/ComponentToken.sol [hunk 1]
```diff
@@ -327,31 +327,23 @@ abstract contract ComponentToken is
         address receiver,
         address controller
     ) public virtual nonReentrant returns (uint256 assets) {
-        if (shares == 0) {
-            revert ZeroAmount();
-        }
-        if (msg.sender != controller) {
-            revert Unauthorized(msg.sender, controller);
-        }
```
Comment (eyqs) [inline nest/src/ComponentToken.sol:335]: Why are we taking away these checks?

File: nest/src/ComponentToken.sol [hunk 2]
```diff
@@ -444,31 +436,22 @@ abstract contract ComponentToken is
         address receiver,
         address controller
     ) public virtual override(ERC4626Upgradeable, IERC7540) nonReentrant returns (uint256 shares) {
-        if (assets == 0) {
-            revert ZeroAmount();
-        }
-        if (msg.sender != controller) {
-            revert Unauthorized(msg.sender, controller);
-        }
```
Comment (eyqs) [inline nest/src/ComponentToken.sol:452]: Same with these checks

File: nest/src/ComponentToken.sol (hunk 1 above)
Comment (ungaro) [reply nest/src/ComponentToken.sol:335]: added them back, sorry my bad.

File: nest/src/ComponentToken.sol (hunk 2 above)
Comment (ungaro) [reply nest/src/ComponentToken.sol:452]: added them back, sorry my bad.

**Review by eyqs — APPROVED** [review]
LGTM

**Pull request merged**